components/button/button-add.js

```diff
@@ -1,20 +1,39 @@
 import '../colors/colors.js';
 import '../icons/icon.js';
 import '../tooltip/tooltip.js';
-import { css, html, LitElement, unsafeCSS } from 'lit';
+import { css, html, LitElement, nothing, unsafeCSS } from 'lit';
+import { VisibleOnAncestorMixin, visibleOnAncestorStyles } from '../../mixins/visible-on-ancestor/visible-on-ancestor-mixin.js';
 import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
 import { getFocusPseudoClass } from '../../helpers/focus.js';
 import { getUniqueId } from '../../helpers/uniqueId.js';
 import { ifDefined } from 'lit/directives/if-defined.js';
 import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
 import { PropertyRequiredMixin } from '../../mixins/property-required/property-required-mixin.js';
+import { styleMap } from 'lit/directives/style-map.js';
+
+const HINT_TIMING = {
+	ALWAYS: 'always',
+	NEARBY: 'nearby',
+	NEVER: 'never'
+};
+const DEFAULT_HINT_DISTANCE = 37;
 
 /**
  * A component for quickly adding items to a specific locaiton.
  */
 class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(LitElement))) {
 	static get properties() {
 		return {
+			/**
+			 * ONLY used when text-visible is FALSE and hint-timing is "nearby". Distance in px where when the user is hovering within that distance the icon will appears. Default is 37.
+			 * @type {number}
+			 */
+			hintNearbyDistance: { type: Number, reflect: true, attribute: 'hint-nearby-distance' },
```

Comment:
I don't love the names of these new properties - let me know other thoughts. Maybe "visibility" instead of "hint"?

Reply:
Ug, yeah, this seems awkward to name. Maybe `iconHintDistance`, `iconShownDistance`, `hoverDistance`, `iconShownHoverDistance`? 

Do we foresee consumers needing to control this? We might get a bunch of inconsistency as a result... if so, do we think that's a concern?